xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -553,6 +555,9 @@ private void handleLdsResponseForConfigUpdate(DiscoveryResponse ldsResponse) {
     List<String> listenerNames = new ArrayList<>(ldsResponse.getResourcesCount());
     try {
       for (com.google.protobuf.Any res : ldsResponse.getResourcesList()) {
+        if (!res.is(Listener.class)) {
```

Comment:
Can you use fully qualified name for types that you want to distinguish in v2 and v3? It's hard to follow as there are some inconsistencies (e.g., constant `ADS_TYPE_URL_LDS` without v3 suffix refers to v2 type while `Listener` refers to v3 type).

Reply:
What will be better is to use v3 unqualified but v2 as qualified (since it will be phased out at some point) wherever possible. So in this case `Listener` is correct but `ADS_TYPE_URL_LDS` should be v3 unqualified.

Reply:
v3 imported classes should be unqualified and v2 ones can be qualified (since they will go away earlier). Constants like `ADS_TYPE_URL_LDS` are defined here: the v3 ones should have no suffix and the existing v2 ones can be renamed with `_V2` suffix for the same reason.

Reply:
Checking type name for `ADS_TYPE_URL_LDS_V2` instead, which might be easier to understand.